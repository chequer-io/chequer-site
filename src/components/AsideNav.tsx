import React from 'react';

export default class AsideNav extends React.Component<any, any> {
  constructor(props) {
    super(props);
  }

  public render() {
    const { style } = this.props;
    return (
      <div className="aside-nav" style={style}>
        <div className={'nav-container'}>
          <a
            href={'https://www.facebook.com/SQLGate/'}
            target={'_blank'}
            className={'item'}
          >
            <span className={'label'}>facebook.com/SQLGate</span>
            <span className="chi-facebook-solid" />
          </a>
          <a
            href={'https://medium.com/chequer'}
            target={'_blank'}
            className={'item'}
          >
            <span className={'label'}>medium.com/chequer</span>
            <span className="chi-medium-solid" />
          </a>
          <a
            href={'https://www.sqlgate.com/'}
            target={'_blank'}
            className={'item'}
          >
            <span className={'label'}>SQLGate.com</span>
            <span className="chi-sqlgate-solid" />
          </a>
        </div>
      </div>
    );
  }
}
