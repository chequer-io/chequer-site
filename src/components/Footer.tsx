import * as React from 'react';
import {Grid} from 'semantic-ui-react';

export class Footer extends React.Component<iPageControlProps, iPageControlState> {
  constructor(props) {
    super(props);

  }

  public render() {
    return (
      <div className='page-footer'>

          <Grid container stackable>
            <Grid.Row>
              <Grid.Column width={8}>
                <div className={'logo'}>
                  <span className='chi-chequer-full' />
                </div>
                <p>
                  Copyright Â© 2018 CHEQUER Inc.<br/>
                  All rights reserved.
                </p>
              </Grid.Column>
              <Grid.Column width={8}>

                <Grid stackable>
                  <Grid.Row>
                    <Grid.Column width={6}>

                      <h3>RECRUIT</h3>
                      <p>
                        <span className="chi-envelope"/> <a href="mailto:join@chequer.io">join@chequer.io</a>
                      </p>

                      <h3>PARTNER</h3>
                      <p>
                        <span className="chi-envelope"/> <a href="mailto:bill@chequer.io">bill@chequer.io</a>
                      </p>

                      <h3>ETC</h3>
                      <p>
                        <span className="chi-envelope"/> <a href="mailto:brant@chequer.io">brant@chequer.io</a>
                      </p>

                    </Grid.Column>
                    <Grid.Column width={10}>

                      <h3>CONTACT INFO</h3>
                      <p>
                        <b>Address</b> : No.1401, 2F, 551-24, Yangcheon-ro, Gangseo-gu, Seoul, Republic of Korea
                      </p>
                      <p><b>TEL</b> : <a href="tel:02-338-1876">02-338-1876</a></p>
                      <p><b>FAX</b> : <a href="tel:02-6008-1876">02-6008-1876</a></p>

                    </Grid.Column>
                  </Grid.Row>
                </Grid>

              </Grid.Column>
            </Grid.Row>
          </Grid>

      </div>
    )
  }
}