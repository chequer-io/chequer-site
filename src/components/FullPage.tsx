import * as React from 'react';
import classNames from 'classnames';

export class FullPage extends React.Component<iFullPageProps, iFullPageState> {
  constructor(props) {
    super(props);

  }

  public render() {
    const divStyleClass = {
      'fullpage': true,
      [this.props.className]: true
    };

    let pageMinHeight = this.props.height;
    if (pageMinHeight < 600) {
      pageMinHeight = 600;
    }

    return (
      <div data-page-index={this.props.pageIndex} className={classNames(divStyleClass)} style={{minHeight: pageMinHeight}}>
        {this.props.children}
      </div>
    )
  }
}