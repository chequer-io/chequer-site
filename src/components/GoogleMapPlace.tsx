import * as React from 'react';

export default class GoogleMapPlace extends React.Component<any, any> {
  constructor(props){
    super(props);

  }
  render() {

    return (
      <div className={'google-map-place'}>
        {this.props.text}
      </div>
    );
  }
}