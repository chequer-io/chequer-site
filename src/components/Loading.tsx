import React from 'react';

const Loading: React.StatelessComponent<{}> = props => {
  return <div>Loading...</div>;
};

export default Loading;
