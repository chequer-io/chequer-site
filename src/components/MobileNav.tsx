import * as React from 'react';
import cx from 'classnames';

export class MobileNav extends React.Component<any, any> {
  constructor(props) {
    super(props);

    this.state = {
      active: false
    }

    this.onClickMenu = this.onClickMenu.bind(this);
  }

  private onClickMenu(e) {
    this.setState({
      active: !this.state.active
    });
  }

  public render() {
    // let {style} = this.props;

    let style = {
      height: '50px'
    };

    if (this.state.active) {
      style.height = this.props.height;
    }

    let menuClassName = {
      open: this.state.active
    };

    return (
      <div className="mobile-nav" style={style}>
        <div className={'nav-header'}>
          <div className={'chequer-logo'}>
            <span className='chi-chequer-full' />
          </div>
          <div className={'right-corner'}>
            <div data-animate-icon="hamburger" className={cx(menuClassName)} onClick={this.onClickMenu}>
              <span />
              <span />
              <span />
              <span />
            </div>
          </div>
        </div>
        <div className={'nav-body'}>

        </div>
      </div>
    )
  }
}