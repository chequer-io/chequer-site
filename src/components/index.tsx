export * from './SideNav';
export * from './AsideNav';
export * from './FullPage';
export * from './PageControl';