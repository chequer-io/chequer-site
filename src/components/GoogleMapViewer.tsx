import React from 'react';
import GoogleMap from 'google-map-react';
import GoogleMapPlace from './GoogleMapPlace';

interface iGoogleMapViewerProps {
  center: number[];
  zoom: number;
  placeCoords: {
    lat: number;
    lng: number;
  };
  text: string;
}

export default class GoogleMapViewer extends React.Component<
  iGoogleMapViewerProps,
  any
> {
  static defaultProps = {
    center: [59.938043, 30.337157],
    zoom: 9,
    placeCoords: { lat: 59.724465, lng: 30.080121 },
  };

  constructor(props) {
    super(props);
  }

  render() {
    const GOOGLE_MAP_API_KEY = 'AIzaSyBVkU-TQbp0YE5hB-WQz3t9huZOISPSKn8';

    return (
      <GoogleMap
        bootstrapURLKeys={{ key: GOOGLE_MAP_API_KEY }}
        center={this.props.center}
        zoom={this.props.zoom}
      >
        <GoogleMapPlace {...this.props.placeCoords} text={this.props.text} />
      </GoogleMap>
    );
  }
}
