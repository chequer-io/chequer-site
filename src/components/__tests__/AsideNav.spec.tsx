import React from 'react';
import { shallow } from 'enzyme';

import { AsideNav } from 'components';

describe('AsideNav', () => {
  let component = null;

  it('renders correctly', () => {
    component = shallow(<AsideNav />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the span sqlgate.com', () => {
    const result = shallow(<AsideNav />).contains(
      <span className={'label'}>SQLGate.com</span>,
    );
    expect(result).toBeTruthy();
  });
});
