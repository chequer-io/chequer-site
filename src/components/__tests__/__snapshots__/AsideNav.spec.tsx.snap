// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideNav matches snapshot 1`] = `
<div
  className="aside-nav"
>
  <div
    className="nav-container"
  >
    <a
      className="item"
      href="https://www.facebook.com/SQLGate/"
      target="_blank"
    >
      <span
        className="label"
      >
        facebook.com/SQLGate
      </span>
      <span
        className="chi-facebook-solid"
      />
    </a>
    <a
      className="item"
      href="https://medium.com/chequer"
      target="_blank"
    >
      <span
        className="label"
      >
        medium.com/chequer
      </span>
      <span
        className="chi-medium-solid"
      />
    </a>
    <a
      className="item"
      href="https://www.sqlgate.com/"
      target="_blank"
    >
      <span
        className="label"
      >
        SQLGate.com
      </span>
      <span
        className="chi-sqlgate-solid"
      />
    </a>
  </div>
</div>
`;
